builds/release/release-core-images.yaml

```diff
@@ -0,0 +1,540 @@
+        echo "Filtering on { product: $product, version: $previous_version }..."
+
+        # The images we build from this pipeline all target .NET Alpine base images, but some are
+        # 'dotnet/aspnet' and some are 'dotnet/runtime'. If this pipeline happens to run, e.g.,
+        # after dotnet/runtime gets updated but before dotnet/aspnet gets updated, then we'll end up
+        # releasing twice in quick succession. To mitigate this problem, we'll wait until *all* the
+        # images/architectures we build in this pipeline are flagged for update.
+        continue=$(jq \
+          --arg p "$product" \
+          --arg v "$previous_version" \
+          --argjson an '["agent","hub","simulated-temperature-sensor","diagnostics"]' \
+          --argjson aa '["amd64","arm32v7","arm64v8"]' '
+            def filter: [
+              $an[] | . as $n | $aa[] | "mcr.microsoft.com/azureiotedge-\($n):\($v)-linux-\(.)"
+            ];
+            [ .[] | select(.product == $p and .version == $v) | .images[] ] | contains(filter)
+          ' $(Pipeline.Workspace)/detect-image-updates/image-updates/updates.json
+        )
+
+        echo "Continue with release? $continue"
+        echo "##vso[task.setvariable variable=continue;isOutput=true]$continue"
+      displayName: Filter image list for core images
+      name: filter
+
+    - task: AzureKeyVault@1
+      displayName: Get secrets
+      condition: and(succeeded(), eq(variables['filter.continue'], 'true'))
+      inputs:
+        azureSubscription: $(az.subscription)
+        keyVaultName: $(kv.name)
+        secretsFilter: TestGitHubAccessToken
+
+    - script: |
+        #!/bin/bash
+        set -euo pipefail
+
+        . scripts/linux/release-tools.sh
+
+        GIT_EMAIL='$(service-account.email)' \
+        BRANCH="${BUILD_SOURCEBRANCH#refs/heads/}" \
+        make_project_release_commit_for_core_image_refresh
+        get_project_release_info
+
+        changelog=$(echo "$OUTPUTS" | jq -rc '.changelog')
+        next=$(echo "$OUTPUTS" | jq -rc '.version')
+        prev=$(echo "$OUTPUTS" | jq -rc '.previous_version')
+        diag_version=$(echo "$OUTPUTS" | jq -rc '.diagnostics_version')
+        tags=$(echo "$OUTPUTS" | jq -rc '.tags')
+
+        echo "##vso[task.setvariable variable=changelog;isOutput=true]$changelog"
+        echo "##vso[task.setvariable variable=version;isOutput=true]$next"
+        echo "##vso[task.setvariable variable=previous_version;isOutput=true]$prev"
+        echo "##vso[task.setvariable variable=diagnostics_version;isOutput=true]$diag_version"
+        echo "##vso[task.setvariable variable=tags;isOutput=true]$tags"
+      displayName: Create a release commit
+      name: commit
+      condition: and(succeeded(), eq(variables['filter.continue'], 'true'))
+      env:
+        GITHUB_TOKEN: "$(TestGitHubAccessToken)"
+
+  - job: ManualTrigger
+    displayName: Manually trigger pipeline to update images
+    condition: eq(variables['Build.Reason'], 'Manual')
```

Comment:
What part of this allows devops to be triggered manually? I couldn't point that part out.

Reply:
This isn't about _allowing_ the pipeline to be triggered manually--in DevOps I think you can trigger any pipeline manually whether it was designed for it or not. This `condition` just passively detects whether the build was triggered manually or not.

I felt it was important to support both types of triggers:
1. [Automatic] Detect-image-updates generates a list of images that need to be updated, and release-core-images wakes up to examine the list, then refreshes/releases the images it is responsible for
2. [Manual] We need to trigger the release manually because:
   1. Detect-image-pipelines is disabled or otherwise not working for some reason
   2. The pipeline ran but encountered an error, so we need to re-run it
   3. Some other reason?